From gadiyar@ti.com  Fri Nov 20 12:45:56 2009
From: Sonic Zhang <sonic.zhang@analog.com>
Date: Mon, 16 Nov 2009 16:19:27 +0530
Subject: USB: musb: error out when anomaly 05000380 is applicable
To: "linux-usb@vger.kernel.org; Greg KH" <greg@kroah.com>
Cc: Sonic Zhang <sonic.zhang@analog.com>, Mike Frysinger <vapier@gentoo.org>, Anand Gadiyar <gadiyar@ti.com>
Message-ID: <1258368569-4323-9-git-send-email-gadiyar@ti.com>


From: Sonic Zhang <sonic.zhang@analog.com>

Since we can't work around anomaly 05000380, throw a build error up and
instruct the user to use a different mode.

Signed-off-by: Sonic Zhang <sonic.zhang@analog.com>
Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Signed-off-by: Anand Gadiyar <gadiyar@ti.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/musb/blackfin.h |   10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/drivers/usb/musb/blackfin.h
+++ b/drivers/usb/musb/blackfin.h
@@ -41,6 +41,16 @@
  *             anomaly - 05000465.
  */
 
+/* The Mentor USB DMA engine on BF52x (silicon v0.0 and v0.1) seems to be
+ * unstable in host mode.  This may be caused by Anomaly 05000380.  After
+ * digging out the root cause, we will change this number accordingly.
+ * So, need to either use silicon v0.2+ or disable DMA mode in MUSB.
+ */
+#if ANOMALY_05000380 && defined(CONFIG_BF52x) && \
+    defined(CONFIG_USB_MUSB_HDRC) && !defined(CONFIG_MUSB_PIO_ONLY)
+# error "Please use PIO mode in MUSB driver on bf52x chip v0.0 and v0.1"
+#endif
+
 #undef DUMP_FIFO_DATA
 #ifdef DUMP_FIFO_DATA
 static void dump_fifo_data(u8 *buf, u16 len)
